src/viam/sdk/module/service.hpp

```diff
@@ -53,6 +53,13 @@ class ModuleService : viam::module::v1::ModuleService::Service {
     /// @param model The model to add.
     void add_model_from_registry(API api, Model model);
 
+    /// @brief Registers a cleanup function to run when the module shuts down. There is
+    /// no default cleanup function and only a single cleanup function can be registered.
+    /// Cleanup functions must be registered before `serve` is called.
```

Comment:
I chose to not allow registration of multiple cleanup functions. Hypothetically, we could allow it, maintain a "stack" of functions, and call them in a FILO order in `~ModuleService`, but I'd rather force module authors to consolidate their cleanup logic into a single function.